test/cli/acceptance/cli_test.go

```diff
@@ -153,18 +167,20 @@ receivers:
 
 	am := amc.Members()[0]
 
-	silence1 := Silence(0, 4).Match("alertname=test1", "severity=warn").Comment("test1")
+	silence1 := Silence(0, 4).Match("test1", "severity=warn").Comment("test1")
 	silence2 := Silence(0, 4).Match("foo").Comment("test foo")
 
```

Comment:
@gotjosh here I plan to also create a silence like:

```go
silence1 := Silence(0, 4).Match("{alertname=test1}", "severity=warn")
```

This will create a silence with the following labels:

```
alertname="{alertname=test3}"
severity=warn
```

We will be able to then see this behaviour break when we use the new fallback parser, as without the additional check, it will create a silence with the following labels instead:

```
alertname=test3"
severity=warn
```

I think this is the right test to write, what do you think? If we're both in agreement I'll add it to the pull request too!